store/unit_store.go

```diff
@@ -89,31 +93,69 @@ func (s unitStores) Refs(f ...RefFilter) ([]*graph.Ref, error) {
 	}
 
 	c_unitStores_Refs_last_numUnitsQueried = 0
-	var allRefs []*graph.Ref
+	var (
+		allRefs   []*graph.Ref
+		allRefsMu sync.Mutex
+	)
+	par := parallel.NewRun(storeFetchPar)
 	for u, us := range uss {
+		u, us := u, us
```

Comment:
Put this line after the `if us == nil { continue }`, as that's better to keep as the topmost statement.